### PR TITLE
feat: Add support for Karpenter v1 controller IAM role permissions

### LINKED ISSUE #3128: [Karpenter] upgrade to v.1.0
Hi,

I have a humble request. Could you please update the code so that we can use Karpenter v.1.0?

Thank you so much!

### PR DESCRIPTION
**Pull request by andy-townsend**
## Description
This PR updates the IAM role for the Karpenter Controller to reflect all the changes made for Karpenter v1.0.0

## Motivation and Context
Without this, karpenter fails to start nodes when upgraded to v1.0.0. Solves errors such as;
```
"error":"tagging nodeclaim, tagging instance, UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:sts::12345678990:assumed-role/karpenter-controller-lab/1723721642521363579 is not authorized to perform: ec2:CreateTags on resource: arn:aws:ec2:eu-west-1:12345678990:instance/i-0d25b72e6a588fa4b because no identity-based policy allows the ec2:CreateTags action. 
```
They removed the `sqs:GetQueueAttributes` [in this change](https://github.com/aws/karpenter-provider-aws/commit/504c71ef5a5867af974182932ea45cc13ea60865)

[This change](https://github.com/aws/karpenter-provider-aws/commit/5e6f896357a16af549db7d32d91105d2020eec17) shows all the changes in the controller IAM policy (taken from website/content/en/docs/getting-started/getting-started-with-karpenter/cloudformation.yaml)

- Closes #3129
- Resolves #3128 

## Breaking Changes
- No

## How Has This Been Tested?

Tested in my lab setup, changes applied and nodes now spin up. 

- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by bryantbiggs** [comment]
I will need to check if this is backwards compatible with Karpenter versions < 1.0 before we can proceed

File: versions.tf [hunk 1]
```diff
@@ -4,15 +4,15 @@ terraform {
...
+      version = ">= 5.60"
     }
     tls = {
       source  = "hashicorp/tls"
       version = ">= 3.0"
     }
     time = {
       source  = "hashicorp/time"
-      version = ">= 0.9"
+      version = ">= 0.7.2"
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply versions.tf:15]: this is a policy only change - version changes should be reverted

[the remark this answers is not in the document]
Comment (andy-townsend) [reply versions.tf]: Apologies, this was for some testing I was doing. The change is only IAM policy related.

**Comment by suraj2410** [comment]
any plans when this would be merged and available for hee?

**Comment by senilio** [comment]
I quickly tried this PR with Karpenter 0.37.0, and I found an issue with launching new nodes.

```
{"level":"ERROR","time":"2024-08-19T05:33:46.765Z","logger":"controller","message":"Reconciler error","commit":"490ef94","controller":"nodeclaim.lifecycle","controllerGroup":"karpenter.sh","controllerKind":"NodeClaim","NodeClaim":{"name":"default-std9a"},"namespace":"","name":"default-std9a","reconcileID":"b6913acf-0fb6-417b-b206-38e3b2bb813f","error":"launching nodeclaim, creating instance, getting launch template configs, getting launch templates, creating launch template, UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:sts::1234567890:assumed-role/KarpenterController-20240816080453288700000001/1724044805759067430 is not authorized to perform: ec2:CreateLaunchTemplate on resource: arn:aws:ec2:eu-west-1:1234567890:launch-template/* because no identity-based policy allows the ec2:CreateLaunchTemplate action.
```

Likely due to the additions of `aws:RequestTag/eks:eks-cluster-name`

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 20.24.0 :tada:

**Pull request merged**